packages/ember-glimmer/tests/integration/helpers/custom-helper-test.js

```diff
@@ -22,4 +27,365 @@ moduleFor('Helpers test: custom helpers', class extends RenderingTest {
     this.assertText('hello world');
   }
 
+  ['@test dashed helper can recompute a new value']() {
+    let destroyCount = 0;
+    let count = 0;
+    let helper;
+
+    this.registerHelper('hello-world', {
+      init() {
+        this._super(...arguments);
+        helper = this;
+      },
+      compute() {
+        return ++count;
+      },
+      destroy() {
+        destroyCount++;
+        this._super();
+      }
+    });
+
+    this.render('{{hello-world}}');
+
+    this.assertText('1');
+
+    this.runTask(() => helper.recompute());
+
+    this.assertText('2');
+
+    assert.strictEqual(destroyCount, 0, 'destroy is not called on recomputation');
+  }
+
+  ['@test dashed helper with arg can recompute a new value']() {
+    let destroyCount = 0;
+    let count = 0;
+    let helper;
+
+    this.registerHelper('hello-world', {
+      init() {
+        this._super(...arguments);
+        helper = this;
+      },
+      compute() {
+        return ++count;
+      },
+      destroy() {
+        destroyCount++;
+        this._super();
+      }
+    });
+
+    this.render('{{hello-world "whut"}}');
+
+    this.assertText('1');
+
+    this.runTask(() => helper.recompute());
+
+    this.assertText('2');
+
+    assert.strictEqual(destroyCount, 0, 'destroy is not called on recomputation');
+  }
+
+  ['@test dashed shorthand helper is called for param changes']() {
+    let computeCount = 0;
+
+    this.registerHelper('hello-world', ([value]) => {
+      computeCount++;
+      return `${value}-value`;
+    });
+
+    this.render('{{hello-world name}}', {
+      name: 'bob'
+    });
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(computeCount, 1, 'compute is called exactly 1 time');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(computeCount, 1, 'compute is called exactly 1 time');
```

Comment:
@zackthehuman yeah, we have plans to make it work. if anything is failing because of that, feel free to skip with @htmlbars for now. I agree with @mixonic that we have some wiggle room on that, but I think we can/should make it work. Just have to circle back on that a bit later.